src/Microsoft.Health.Dicom.Core/Features/Query/QueryService.cs

```diff
@@ -63,6 +76,53 @@ public class QueryService : IQueryService
             return new QueryResourceResponse(responseMetadata);
         }
 
+        private Dictionary<QueryResource, HashSet<CustomTagFilterDetails>> GenerateQueryResourceToCustomTagMapping(IEnumerable<CustomTagStoreEntry> customTags)
```

Comment:
Can we move this to CustomTagService. This is very specific to custom tag, it does not belong here.

Reply:
It's awkward to move since it describes things by QueryResource...There is also currently no singular CustomTagService! It's broken out by action. I can create a separate extension class if you still fell like it's too specific!

---
In reply to: [582550008](https://github.com/microsoft/dicom-server/pull/567#discussion_r582550008) [](ancestors = 582550008)